### ISSUE TITLE
Upgrade Kustomize version from 3.8.2 to 3.8.10

### ISSUE THREAD

**Issue by ron1** [rancher/fleet#287]
Upgrade bundled Kustomize version from 3.8.2 to 3.8.10 to incorporate numerous bug fixes between August 2020 and February 2021. Note that this version continues to disable kyaml by default. The affected libraries are as follows:

Kustomize 3.8.2
```
sigs.k8s.io/cli-utils v0.16.0
sigs.k8s.io/kustomize/api v0.6.0
sigs.k8s.io/kustomize/kyaml v0.7.0 (v0.7.1)
```

Kustomize 3.8.10
```
sigs.k8s.io/cli-utils v0.23.1
sigs.k8s.io/kustomize/api v0.6.9
sigs.k8s.io/kyaml v0.10.7
```

**Comment by cbron** [comment]
This should be linked with https://github.com/rancher/tasks/issues/158

**Comment by nickgerace** [comment]
Potential blocker: https://github.com/kubernetes-sigs/kustomize/issues/3446

**Comment by nickgerace** [comment]
Slightly changed dependencies based on: https://github.com/kubernetes-sigs/kustomize/blob/kustomize/v3.8.10/kustomize/go.mod

**Comment by nickgerace** [comment]
This is awaiting a new tag and chart.

**Comment by StrongMonkey** [comment]
For QA: Also make sure our kustomize example in rancher/fleet-examples still works and no regression bug.

**Comment by jiaqiluo** [comment]
The fix is validated on rancher:`v2.5.8-rc4` with fleet:`0.3.500-rc05` 

Fleet works as expected, no issue is found. 
Both the single-cluster and multi-cluster kustomize example in rancher/fleet-examples works as expected.
